atlasdb-tests-shared/src/main/java/com/palantir/timelock/paxos/InMemoryTimelockExtension.java

```diff
@@ -0,0 +1,243 @@
+import com.palantir.lock.client.RequestBatchersFactory;
+import com.palantir.lock.client.TransactionStarter;
+import com.palantir.lock.v2.NamespacedTimelockRpcClient;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.sls.versions.OrderableSlsVersion;
+import com.palantir.timelock.config.ClusterInstallConfiguration;
+import com.palantir.timelock.config.ImmutableClusterInstallConfiguration;
+import com.palantir.timelock.config.ImmutableDefaultClusterConfiguration;
+import com.palantir.timelock.config.ImmutableSqlitePaxosPersistenceConfiguration;
+import com.palantir.timelock.config.ImmutableTimeLockRuntimeConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration.PaxosLeaderMode;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
+import com.palantir.timestamp.ManagedTimestampService;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampService;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public final class InMemoryTimelockExtension
+        implements BeforeEachCallback, AfterEachCallback, TimeLockServices, Closeable {
+    private static final String USER_AGENT_NAME = "user-agent";
+    private static final String USER_AGENT_VERSION = "3.1415926.5358979";
+    private static final UserAgent USER_AGENT = UserAgent.of(UserAgent.Agent.of(USER_AGENT_NAME, USER_AGENT_VERSION));
+    private static final int THREAD_POOL_SIZE = 128;
+    private static final int BLOCKING_TIMEOUT_MS = 60 * 800; // 0.8 mins to ms
+
+    private String client;
+    private TimeLockAgent timeLockAgent;
+    private TimeLockServices delegate;
+    private TimeLockHelperServices helperServices;
+
+    private LockLeaseService lockLeaseService;
+    private NamespacedConjureTimelockServiceImpl namespacedConjureTimelockService;
+
+    public InMemoryTimelockExtension() {
+        this.client = "client";
+    }
+
+    void setClient(String client) {
+        this.client = client;
+    }
+
+    @Override
+    public void beforeEach(ExtensionContext var1) {
+        PaxosInstallConfiguration paxos = PaxosInstallConfiguration.builder()
+                .dataDirectory(tryCreateSubFolder())
+                .leaderMode(PaxosLeaderMode.SINGLE_LEADER)
+                .sqlitePersistence(ImmutableSqlitePaxosPersistenceConfiguration.builder()
+                        .dataDirectory(tryCreateSubFolder())
```

Comment:
We used to name the subfolder with `this.client` in here. I didn't understand the particular reason. Anyway, `ProfileStoreTest` is happy even though we don't name the subfolder with the client. In the future, when we migrate other tests to use `InMemoryTimelockExtension`, we may encounter with problems due to this. Something to keep in mind.